src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -400,19 +400,39 @@ public void OnCompleted(Func<object, Task> callback, object state)
 
         public void Flush()
         {
-            ProduceStartAndFireOnStarting().GetAwaiter().GetResult();
+            var produceStartTask = ProduceStartAndFireOnStarting();
+            if (produceStartTask.Status != TaskStatus.RanToCompletion)
+            {
+                produceStartTask.GetAwaiter().GetResult();
```

Comment:
Is this pattern measurably faster? The awaiter is a single field stack allocated struct and GetResult() no-ops after a single if condition like the one above.